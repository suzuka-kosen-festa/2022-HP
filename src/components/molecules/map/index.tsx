import React from "react";
import tw from "twin.macro";
import type { MapProperties } from "../../../models";

const GOOGLE_MAP_API_KEY = process.env.GOOGLE_MAP_API_KEY || "";

const MapContainer = tw.figure`w-full lg:w-6/12`

const MapWrapper = tw.div`relative w-full h-0 pt-[100%] lg:pt-[56.25%]`

const Map: React.FC<MapProperties> = ({ center, title, zoom, ...rest }) => (
  <MapContainer>
    <MapWrapper>
      <iframe
        id={"map"}
        title={`${title}の地図`}
        role={"document"}
        src={`https://www.google.com/maps/embed/v1/view?key=${GOOGLE_MAP_API_KEY}&center=${center.lat}%2C${center.lng}&zoom=${zoom}`}
        loading={"lazy"}
        referrerPolicy={"no-referrer-when-downgrade"}
        css={tw`absolute top-0 left-0 w-full h-full`}
        allowFullScreen={true}
        {...rest}
      ></iframe>
    </MapWrapper>
  </MapContainer>
);

export { Map };
