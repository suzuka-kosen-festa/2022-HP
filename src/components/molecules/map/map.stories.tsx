import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Map } from ".";

type T = typeof Map;
type Story = ComponentStoryObj<T>;

const data = {
  center: {
    lat: 34.851_671_242_413_17,
    lng: 136.581_323_097_843_55,
  },
  title: "鈴鹿工業高等専門学校",
  zoom: 15,
};

export default {
  args: { center: data.center, title: data.title, zoom: data.zoom },
  component: Map,
} as ComponentMeta<T>;

export const Default: Story = {};
