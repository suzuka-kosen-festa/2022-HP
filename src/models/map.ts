import type { ComponentProps, ReactHTML } from "react";

export type MapProperties = Omit<ComponentProps<ReactHTML["iframe"]>, "children"> & {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  center: google.maps.LatLngLiteral;
  title: string;
  zoom: number;
};
