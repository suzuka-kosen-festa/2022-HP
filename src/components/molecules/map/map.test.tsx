import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./map.stories";

const { Default } = composeStories(stories);

const figureoptions = {
  name: ""
}

const documentoptions = {
  name: "鈴鹿工業高等専門学校の地図"
}

describe("(components) molecules/map", () => {
  test("to be molecules", () => {
    const { container } = render(<Default />);
    expect(container).toBeMolecule();
  });
  test("to be [role=figure]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("figure", figureoptions))
  })
  test("to be [role=document]", () => {
    const { getByRole } = render(<Default />)
    expect(getByRole("document", documentoptions))
  })
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
