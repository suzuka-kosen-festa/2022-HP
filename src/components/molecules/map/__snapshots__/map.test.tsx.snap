// Vitest Snapshot v1

exports[`(components) molecules/map > take snap shot 1`] = `
<div>
  <figure
    class="css-s3afwo"
  >
    <div
      class="css-1yuaof5"
    >
      <iframe
        allowfullscreen=""
        class="css-u7aauf"
        id="map"
        loading="lazy"
        referrerpolicy="no-referrer-when-downgrade"
        role="document"
        src="https://www.google.com/maps/embed/v1/view?key=AIzaSyCk9WMilIT2wiBMqcn1YZQKJz5Lzy9ipQg&center=34.85167124241317%2C136.58132309784355&zoom=15"
        title="鈴鹿工業高等専門学校の地図"
      />
    </div>
  </figure>
</div>
`;
